lms/djangoapps/mobile_api/users/views.py

```diff
@@ -530,6 +533,121 @@ def my_user_info(request, api_version):
+        about each user's enrolment status.
+
+    **Example Response**
+
+        ```json
+        [
+            {
+                "course_id": "course-v1:a+a+a",
+                "course_name": "a",
+                "is_active": true
+            },
+            {
+                "course_id": "course-v1:b+b+b",
+                "course_name": "b",
+                "is_active": true
+            },
+            {
+                "course_id": "course-v1:c+c+c",
+                "course_name": "c",
+                "is_active": false
+            },
+            ...
+        ]
+        ```
+    """
+    def get(self, request, *args, **kwargs) -> Response:
+        """
+        Gets user's enrollments status.
+        """
+        active_status_date = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=30)
+        username = kwargs.get('username')
+        course_ids_where_user_has_completions = self._get_course_ids_where_user_has_completions(
+            username,
+            active_status_date,
+        )
+        enrollments_status = self._build_enrollments_status_dict(
+            username,
+            active_status_date,
+            course_ids_where_user_has_completions
+        )
+        return Response(enrollments_status)
+
+    def _build_enrollments_status_dict(
+        self,
+        username: str,
+        active_status_date: datetime,
+        course_ids: List[str],
+    ) -> List[Dict[str, bool]]:
+        """
+        Builds list with dictionaries with user's enrolments statuses.
+        """
+        user = get_object_or_404(User, username=username)
+        user_enrollments = CourseEnrollment.enrollments_for_user(user).select_related('course')
+        mobile_available = [
+            enrollment for enrollment in user_enrollments
+            if is_mobile_available_for_user(user, enrollment.course_overview)
+        ]
+        enrollments_status = []
+        for user_enrollment in mobile_available:
+            course_id = str(user_enrollment.course_overview.id)
+            enrollments_status.append(
+                {
+                    'course_id': course_id,
+                    'course_name': user_enrollment.course_overview.display_name,
+                    'is_active': bool(
+                        course_id in course_ids
+                        or user_enrollment.created > active_status_date
+                    )
+                }
+            )
+        return enrollments_status
+
+    @staticmethod
+    def _get_course_ids_where_user_has_completions(
+        username: str,
+        active_status_date: datetime,
+    ) -> List[str]:
+        """
+        Gets course ids where user has completions.
+        """
+        user_completions_last_month = BlockCompletion.objects.filter(
+            user__username=username,
+            created__gte=active_status_date
+        )
+        return [str(completion.block_key.course_key) for completion in user_completions_last_month]
```

Comment:
Each completion also stores the course key so it would be more efficient to directly query that as follows:

```context_keys = BlockCompletion.objects.filter(
            user__username=username,
            created__gte=active_status_date
        ).values_list('context_key', flat=True).distinct()
```

Additionally, a user might have dozens or even hundreds of completions in a course, one for each block each for the same course so that above would have a lot of duplicates.